### ISSUE TITLE
Code Review: Refactoring and Methodological Improvements

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#13]
**Correctness Fixes**
- [ ]  Fix distributed MoE load balancing calculation (remove multiplication by total_tokens before all-reduce).
- [ ]  Add a safeguard to ensure n_clusters >= 1 in KMeans for false negative clustering.
- [ ]  Handle anchors with fewer than k_negatives (e.g. via padding or per-batch k)

**Code Organization & Readability**
- [ ]  Refactor streaming_dataset.py into modular steps: (1) triplet filtering, (2) margin logic, (3) sampling.
- [ ]  Add inline comments explaining excluded, margin weighting, and negative sampling strategy.
- [ ]  Remove unused code, e.g., unused key in generator, unused config values.

**Performance & Efficiency**
- [ ]  Optimize streaming_dataset.py to avoid loading full triplet list in each worker (e.g., shared memory or partitioned caches).
- [ ]  Consider consolidating NAICS triplet files to reduce Polars I/O overhead.
- [ ]  Reassess the value of persistent worker DataLoader vs. static precomputed dataset.

**Methodology Enhancements**

- [ ]  Implement [Decoupled Contrastive Learning (DCL)](https://arxiv.org/pdf/2110.06848) loss:
```
pos_term = -pos_sim
neg_term = logsumexp(neg_sims)
loss = (pos_term + neg_term).mean()
```
- [ ]  Retain false negative masking (-inf on known false negatives).
- [ ]  Log loss carefully—DCL loss may yield negative values.
- [ ]  Optionally, experiment with DCLW (weighted variant), using weighting on hard positives.

**Training Setup**
- [ ]  Remove unused warmup_steps config if not applying warmup scheduler.
- [ ]  Optionally add warmup + cosine decay if large training jobs benefit from it.
